extensions/blocks/instagram-gallery/image-transition.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, useRef } from '@wordpress/element';
+import { Animate } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+export default function ImageTransition( {
+	src,
+	attributes
+} ) {
+	const [ loaded, setLoaded ] = useState( false );
+	const [ size, setSize ] = useState( 'auto' );
+	const { columns, spacing } = attributes;
+	const transitionSpeed = '.3';
+	const img = useRef();
+	let componentMounted = false;
+
+	const onImageLoad = () => {
+		if ( componentMounted ) {
+			setLoaded( true );
+		}
+	};
+
+	useEffect( () => {
+		const imgSrc = src;
+		if ( imgSrc ) {
+			// First load image as a new window.Image and then update local state when it is loaded,
+			// this lets us handle the fade in animation
+			const tmpImg = new window.Image();
+			tmpImg.src = imgSrc;
+			tmpImg.onload = onImageLoad;
+			componentMounted = true;
+		}
+		if ( img.current ) {
+			setSize( img.current.parentNode.getBoundingClientRect().width );
+		}
+	}, [ columns, spacing ] );
+
+	const imageLoadedStyle = {
+		opacity: '1',
+		height: size,
+		width: size,
+	};
+
+	const imageLoadedLoadingStyle = {
+		opacity: '0',
+	};
+
+	const imageStyle = {
+		opacity: '0',
+		height: '0',
+		width: '0',
+		position: 'absolute',
+		top: '0',
+		left: '0',
+		'z-index': '10',
+	};
+
+	imageStyle.transition = `opacity ${ transitionSpeed }s ease-in`;
+
+	const loadingStyle = {
+		opacity: '1',
+		height: size,
+		width: size,
+		position: 'relative',
+		top: '0',
+		left: '0',
+	};
+
+	loadingStyle.transition = `opacity ${ transitionSpeed }s ease-out`;
+
+	return (
+		<>
+			<Animate type="loading">
```

Comment:
[According to the docs](https://developer.wordpress.org/block-editor/components/animate/#loading), this should be removed when loading has finished. If we bring the `loaded` check out, would that mean we'd need to do less styling of the image based on the state and only include the placeholder when needed?

I realise that might make for an abrupt end to the loading animation, but could be worth investigating.